src/Mobs/Path.cpp

```diff
@@ -0,0 +1,382 @@
+#include "Globals.h"
+#ifndef COMPILING_PATHFIND_DEBUGGER
+/* MCServer headers */
+#include "../World.h"
+#include "../Chunk.h"
+#endif
+
+
+#include "Path.h"
+
+#include <stdio.h>
+#include <exception>
+#include <cmath>
+
+#define DISTANCE_MANHATTEN 0  // 1: More speed, a bit less accuracy 0: Max accuracy, less speed.
+#define HEURISTICS_ONLY 0  // 1: Much more speed, much less accurate.
+#define CALCULATIONS_PER_STEP 5  // Higher means more CPU load but faster path calculations.
+// The only version which guarantees the shortest path is 0, 0.
+
+enum class eCellStatus {OPENLIST,  CLOSEDLIST,  NOLIST};
+struct cPathCell
+{
+	Vector3d m_Location;   // Location of the cell in the world
+	int m_F, m_G, m_H;     // F, G, H as defined in regular A*.
+	eCellStatus m_Status;  // Which list is the cell in? Either non, open, or closed.
+	cPathCell * m_Parent;  // Cell's parent, as defined in regular A*.
+	bool m_IsSolid;	       // Is the cell an air or a solid? Partial solids are currently considered solids.
+};
+
+
+
+
+
+bool compareHeuristics::operator()(cPathCell * & a_V1, cPathCell * & a_V2)
+{
+	return a_V1->m_F > a_V2->m_F;
+}
+
+
+
+
+
+/* cPath implementation */
```

Comment:
Move constructor to the top of implementation, then destructor, then the rest. The order should match the declaration order in the header file.